### ISSUE TITLE
Specifying --files from sbt errors because semanticdb not found

### ISSUE THREAD

**Issue by Katrix** [scalacenter/scalafix#1874]
Trying to specify `--files` for a task and pointing it to a specific file errrors with a message saying it can't find the semanticdb. Removing the `--files` argument succeeds like normal.

This errors:
`coreSharedProj/scalafixAll --files=coreShared\\src\\main\\scala\\core\\controllers\\IOController.scala`

This works like normal:
`coreSharedProj/scalafixAll`

**Comment by bjaglin** [comment]
Thanks for the report!

What you are seeing is most likely due to the current behavior of `scalafixAll`, which runs `scalafix` several times under the hood, once per known configuration, after compiling source files in each of them to generate semanticdb information. Based on the `\\main` path, the targeted file is in the `Compile` configuration where the run most likely succeeds, but not in the `Test` one where the semanticdb information is not available as the file does not belong to that configuration.

In other words, `--files` option only makes sense on a local `scalafix` invocation. It's really far from obvious so sbt-scalafix should prevent doing otherwise. I'll work on a PR to make it impossible to use that parameter on `scalafixAll` and on build-level/aggregating `scalafix` invocations.

**Comment by bjaglin** [comment]
Guard-rails added, feel free to test it and report back any improvement idea.
```scala
resolvers ++= Resolver.sonatypeOssRepos("snapshots")
addSbtPlugin("ch.epfl.scala" % "sbt-scalafix" % "0.11.1+16-f27ea8b5-SNAPSHOT")
```
